Wikipedia WebM sample has wrong aspect ratio

jernoble opened the pull request:
#### 29decfdc29bb1f47af0e2b463f0a19316ef7e7b2
<pre>
Wikipedia WebM sample has wrong aspect ratio
<a href="https://bugs.webkit.org/show_bug.cgi?id=282555">https://bugs.webkit.org/show_bug.cgi?id=282555</a>
<a href="https://rdar.apple.com/139229383">rdar://139229383</a>

Reviewed by Jean-Yves Avenard.

285809@main fixed aspect ratio of webm files whose source pixels were 1:1,
but the resulting pixel-aspect-ratio setting was incorrect for files with
non-1:1 sources and a sample-aspect-ratio setting.

* LayoutTests/media/content/test-par-16-9.webm: Added.
* LayoutTests/media/media-vp9-dar-webm-expected.txt:
* LayoutTests/media/media-vp9-dar-webm.html:
* LayoutTests/media/media-vp9-sar-webm-expected.txt: Added.
* LayoutTests/media/media-vp9-sar-webm.html: Added.
* Source/WebCore/platform/graphics/cocoa/CMUtilities.mm:
(WebCore::createFormatDescriptionFromTrackInfo):

Canonical link: <a href="https://commits.webkit.org/286898@main">https://commits.webkit.org/286898@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6a52f5f6d854bf439f01c16f9edd9e7e2eb43032

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/77453 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/56488 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/30369 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/82034 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/28735 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/79570 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/65636 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/4785 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/60694 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18698 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/80520 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/50663 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/66478 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40975 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/48065 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23975 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/27058 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/69174 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/24312 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/83442 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/4833 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/3283 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/68938 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/4989 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/66446 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/68202 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17035 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/12203 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/10298 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/4780 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/7595 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/4799 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/8234 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/6558 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebCore/platform/graphics/cocoa/CMUtilities.mm:
```diff
@@ -210,12 +210,11 @@ static CFStringRef convertToCMYCbCRMatrix(PlatformVideoMatrixCoefficients coeffi
             CFDictionaryAddValue(extensions.get(), kCVImageBufferYCbCrMatrixKey, cmMatrix.get());
     }
     if (videoInfo.size != videoInfo.displaySize) {
```
We can remove the `#import <numeric>` at the top if we don't use gcd

jyavenard commented on LayoutTests/media/media-vp9-sar-webm.html:
```diff
@@ -0,0 +1,23 @@
+        findMediaElement();
+        run('video.src = "content/test-par-16-9.webm"');
+        await waitFor(video, 'loadedmetadata');
+        testExpected('video.videoWidth', 320);
```
I don't think this test is sufficient to check that we properly render the AR.
videoWidth/videoHeight would have been correct before, they would just have been rendered incorrectly.

A reftest may be required here, or checking the width/height as well (not verified)

jyavenard reviewed (approved):
r+ with extra test to verify the rendered size is also correct

jernoble committed:
Wikipedia WebM sample has wrong aspect ratio
https://bugs.webkit.org/show_bug.cgi?id=282555
rdar://139229383

Reviewed by Jean-Yves Avenard.

285809@main fixed aspect ratio of webm files whose source pixels were 1:1,
but the resulting pixel-aspect-ratio setting was incorrect for files with
non-1:1 sources and a sample-aspect-ratio setting.

* LayoutTests/media/content/test-par-16-9.webm: Added.
* LayoutTests/media/media-vp9-dar-webm-expected.txt:
* LayoutTests/media/media-vp9-dar-webm.html:
* LayoutTests/media/media-vp9-sar-webm-expected.txt: Added.
* LayoutTests/media/media-vp9-sar-webm.html: Added.
* Source/WebCore/platform/graphics/cocoa/CMUtilities.mm:
(WebCore::createFormatDescriptionFromTrackInfo):

Canonical link: https://commits.webkit.org/286898@main
